testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -125,6 +126,10 @@ def test_update_data_fail(self, results):
         with pytest.raises(AttributeError, match=msg):
             results.update({"data": 0})
 
+    def test_pickle(self, results):
```

Comment:
Would it be worth adding a test similar to what was being done in #3309 (probably in `test_multiprocessing`), so that we know that it'll work fine?